core/services/workflows/v2/capability_executor.go

```diff
@@ -30,8 +50,33 @@ func (c *ExecutionHelper) CallCapability(ctx context.Context, request *sdkpb.Cap
 	}
 	defer func() { <-c.capCallsSemaphore }()
 
+	// TODO: if this needs to be extended to cover more extensions, consider
+	// something more elegant like a middleware layer or a handler dispatcher.
```

Comment:
For now, maybe simply have one execution helper with a boolean flag in the constructor "allowCapabilityCalls"?